src/query/compiler.js

```diff
@@ -678,12 +678,14 @@ assign(QueryCompiler.prototype, {
     if (!withs) return;
     const sql = [];
     let i = -1;
+    let recursive = '';
     while (++i < withs.length) {
       const stmt = withs[i];
+      if (stmt.recursive) recursive = 'recursive ';
```

Comment:
This looks a bit redundant, you are resetting string multiple times and constructing string at this time looks kinda premature. Might be cleaner to set boolean flag instead and do something like

```
const withBlock = isRecursive ? : 'with recursive ' : 'with '
```
later on